src/protocols/secure_channel/tests/TestCASESession.cpp

```diff
@@ -874,6 +864,241 @@ TEST_F(TestCASESession, Sigma1ParsingTest)
     TestSigma1Parsing(mem, bufferSize, Sigma1SessionIdTooBig);
 }
 
+TEST_F(TestCASESession, EncodeSigma1Test)
+{
+    CASESession::EncodeSigma1Inputs encodeParams;
+
+    uint8_t random[32];
+    EXPECT_EQ(chip::Crypto::DRBG_get_bytes(&random[0], 32), CHIP_NO_ERROR);
+    encodeParams.initiatorRandom    = ByteSpan(random, sizeof(random));
+    encodeParams.initiatorSessionId = 7315;
+    uint8_t destinationId[32]       = { 0xDE, 0xAD };
+    encodeParams.destinationId      = ByteSpan(destinationId, sizeof(destinationId));
+
+    ReliableMessageProtocolConfig MRPConfig = GetDefaultMRPConfig();
+    encodeParams.initiatorMrpConfig         = &MRPConfig;
+
+    {
+        System::PacketBufferHandle msg;
+        // EncodeSigma1 should fail when there is no public key
+        EXPECT_EQ(CHIP_ERROR_INCORRECT_STATE, EncodeSigma1(msg, encodeParams));
```

Comment:
EncodeSigma1 should probably be static, if it does not touch members, which would make it clear why this is OK without a new CASESession instance?